shm/include/store.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "cargo.hpp"
+#include "player.hpp"
+
+#include <memory>
+#include <vector>
+
+class Store {
+public:
+    enum class Response { done, lack_of_money, lack_of_cargo, lack_of_space };
+
+    Store(const std::vector<std::shared_ptr<Cargo>>& stock);
+
+    Response buy(Cargo* cargo, Player* player);
+    Response sell(Cargo* cargo, Player* player);
+
+private:
+    std::vector<std::shared_ptr<Cargo>> stock_;
+
+    void load(std::shared_ptr<Cargo> cargo);
+    void unload(Cargo* cargo);
```

Comment:
The same question as in Player class - why would you use shared_ptr in load function argument and raw ptr in unload func?

Reply:
Here I keep the same convenction like above.